src/main/java/nl/hsac/fitnesse/fixture/util/HttpClient.java

```diff
@@ -130,17 +133,35 @@ protected void getResponse(String url, HttpResponse response, HttpRequestBase me
             org.apache.http.HttpResponse resp;
             CookieStore store = response.getCookieStore();
             if (store == null) {
-                resp = getHttpResponse(url, method);
-            } else {
-                resp = getHttpResponse(store, url, method);
+            	store = new BasicCookieStore();
+//                resp = getHttpResponse(url, method);
             }
+//            } else {
+                resp = getHttpResponse(store, url, method);
+//            }
             int returnCode = resp.getStatusLine().getStatusCode();
             response.setStatusCode(returnCode);
             HttpEntity entity = resp.getEntity();
 
             Map<String, String> responseHeaders = response.getResponseHeaders();
             for (Header h : resp.getAllHeaders()) {
+            	if(h.getName().equalsIgnoreCase("Set-Cookie"))
```

Comment:
Does this not cause too many values for 'Set-Cookie' in responseHeader when response's cookieStore was filled before request? 
I believe it will now appear as-if all cookies present in response's cookieStore were sent by server (even if they were not but added programmatically before the request was sent). That is no good. The responseHeaders should only contain those Set-Cookies actually sent by the server.


**If that is no problem** : would it not be nicer to:
- first process all headers as before. 
- And then (if `responseHeaders` contains 'Set-Cookie') put all cookies in `store` in the responseHeaders for key 'Set-Cookie'.

Basically the same as you do now, but after the for loop instead of nested inside it?